tools/deployment/test_torchserver.py

```diff
@@ -21,52 +26,94 @@ def parse_args():
         '--device', default='cuda:0', help='Device used for inference')
     parser.add_argument(
         '--score-thr', type=float, default=0.5, help='bbox score threshold')
+    parser.add_argument(
+        '--workdir',
+        type=str,
+        default=None,
+        help='output directory to save drawn results.')
     args = parser.parse_args()
     return args
 
 
-def parse_result(input, model_class):
-    bbox = []
-    label = []
-    score = []
-    for anchor in input:
-        bbox.append(anchor['bbox'])
-        label.append(model_class.index(anchor['class_name']))
-        score.append([anchor['score']])
-    bboxes = np.append(bbox, score, axis=1)
-    labels = np.array(label)
-    result = bbox2result(bboxes, labels, len(model_class))
-    return result
+def align_ts_output(inputs, metainfo, device):
+    bboxes = []
+    labels = []
+    scores = []
+    for i, pred in enumerate(inputs):
+        bboxes.append(pred['bbox'])
+        labels.append(pred['class_label'])
+        scores.append(pred['score'])
+    pred_instances = InstanceData(metainfo=metainfo)
+    pred_instances.bboxes = torch.tensor(
+        bboxes, dtype=torch.float32, device=device)
+    pred_instances.labels = torch.tensor(
+        labels, dtype=torch.int64, device=device)
+    pred_instances.scores = torch.tensor(
+        scores, dtype=torch.float32, device=device)
+    ts_data_sample = DetDataSample(pred_instances=pred_instances)
+    return ts_data_sample
 
 
 def main(args):
+    register_all_modules()
     # build the model from a config file and a checkpoint file
     model = init_detector(args.config, args.checkpoint, device=args.device)
     # test a single image
-    model_result = inference_detector(model, args.img)
-    for i, anchor_set in enumerate(model_result):
-        anchor_set = anchor_set[anchor_set[:, 4] >= 0.5]
-        model_result[i] = anchor_set
+    results = inference_detector(model, args.img)
+    metainfo = results.pred_instances.metainfo
+    pred_instances = InstanceData(metainfo=metainfo)
+
+    indices = results.pred_instances.scores >= args.score_thr
+    pred_instances.scores = results.pred_instances.scores[indices]
+    pred_instances.labels = results.pred_instances.labels[indices]
+    pred_instances.bboxes = results.pred_instances.bboxes[indices, :]
+    pytorch_results = DetDataSample(pred_instances=pred_instances)
```

Comment:
You can set score_thr at `visualizer.add_datasample()`
```suggestion
    pytorch_results = inference_detector(model, args.img)
```